src/overlays/actors/ovl_En_GirlA/z_en_girla.c

```diff
@@ -93,200 +93,251 @@ static char* D_80A3C590[] = {
+      ItemGive_DekuSeeds, BuyEvent_ShieldDiscount },
+    /* 34 */
+    { OBJECT_GI_TRUTH_MASK, GID_MASK_TRUTH, func_80A3C498, 0, 1, 0x70AF, 0x70C3, GI_MASK_TRUTH, CanBuy_DekuSeeds,
+      ItemGive_DekuSeeds, BuyEvent_ShieldDiscount },
+    /* 35 */
+    { OBJECT_GI_ZORAMASK, GID_MASK_ZORA, NULL, 0, 1, 0x70B9, 0x70C1, GI_MASK_ZORA, CanBuy_DekuSeeds, ItemGive_DekuSeeds,
+      BuyEvent_ShieldDiscount },
+    /* 36 */
+    { OBJECT_GI_GOLONMASK, GID_MASK_GORON, NULL, 0, 1, 0x70B8, 0x70C0, GI_MASK_GORON, CanBuy_DekuSeeds,
+      ItemGive_DekuSeeds, BuyEvent_ShieldDiscount },
+    /* 37 */
+    { OBJECT_GI_GERUDOMASK, GID_MASK_GERUDO, NULL, 0, 1, 0x70BA, 0x70C2, GI_MASK_GERUDO, CanBuy_DekuSeeds,
+      ItemGive_DekuSeeds, BuyEvent_ShieldDiscount },
+    /* 38 */
+    { OBJECT_GI_SOLDOUT, GID_SOLDOUT, func_8002EBCC, 0, 0, 0x00BD, 0x70C2, GI_MASK_GERUDO, CanBuy_SoldOut, NULL, NULL },
+    /* 39 */
+    { OBJECT_GI_FIRE, GID_BLUE_FIRE, func_8002EBCC, 300, 1, 0x00B9, 0x00B8, GI_BLUE_FIRE, CanBuy_BlueFire,
+      ItemGive_BottledItem, BuyEvent_ShieldDiscount },
+    /* 40 */
+    { OBJECT_GI_INSECT, GID_BUG, func_80A3C498, 50, 1, 0x00BB, 0x00BA, GI_BUGS, CanBuy_Bugs, ItemGive_BottledItem,
+      BuyEvent_ShieldDiscount },
+    /* 41 */
+    { OBJECT_GI_GHOST, GID_BIG_POE, func_80A3C498, 50, 1, 0x506F, 0x5070, GI_BIG_POE, CanBuy_Poe, ItemGive_BottledItem,
+      BuyEvent_ShieldDiscount },
+    /* 42 */
+    { OBJECT_GI_GHOST, GID_POE, func_80A3C498, 30, 1, 0x506D, 0x506E, GI_POE, CanBuy_Poe, ItemGive_BottledItem,
+      BuyEvent_ShieldDiscount },
+    /* 43 */
+    { OBJECT_GI_SOUL, GID_FAIRY, func_80A3C498, 50, 1, 0x00B7, 0x00B6, GI_FAIRY, CanBuy_Fairy, ItemGive_BottledItem,
+      BuyEvent_ShieldDiscount },
+    /* 44 */
+    { OBJECT_GI_ARROW, GID_ARROWS_SMALL, func_8002EBCC, 20, 10, 0x00A0, 0x008A, GI_ARROWS_SMALL, CanBuy_Arrows,
+      ItemGive_Arrows, BuyEvent_ShieldDiscount },
+    /* 45 */
+    { OBJECT_GI_BOMB_1, GID_BOMB, func_8002EBCC, 80, 20, 0x001C, 0x0006, GI_BOMBS_20, CanBuy_Bombs, ItemGive_Bombs,
+      BuyEvent_ShieldDiscount },
+    /* 46 */
+    { OBJECT_GI_BOMB_1, GID_BOMB, func_8002EBCC, 120, 30, 0x001D, 0x001E, GI_BOMBS_30, CanBuy_Bombs, ItemGive_Bombs,
+      BuyEvent_ShieldDiscount },
+    /* 47 */
+    { OBJECT_GI_BOMB_1, GID_BOMB, func_8002EBCC, 35, 5, 0x00CB, 0x00CA, GI_BOMBS_5, CanBuy_Bombs, ItemGive_Bombs,
+      BuyEvent_ShieldDiscount },
+    /* 48 */
+    { OBJECT_GI_LIQUID, GID_POTION_RED, func_8002EBCC, 40, 1, 0x0064, 0x0062, GI_POTION_RED, CanBuy_RedPotion,
+      ItemGive_BottledItem, BuyEvent_ShieldDiscount },
+    /* 49 */
+    { OBJECT_GI_LIQUID, GID_POTION_RED, func_8002EBCC, 50, 1, 0x0065, 0x0063, GI_POTION_RED, CanBuy_RedPotion,
+      ItemGive_BottledItem, BuyEvent_ShieldDiscount },
 };
 
-static s16 D_80A3CCB4[8] = { 0x0005, 0x000A, 0x000F, 0x0014, 0x0019, 0x001E, 0x0023, 0x0028 };
+// Defines the Hylian Shield discount amount
+static s16 randomDiscount[] = { 5, 10, 15, 20, 25, 30, 35, 40 };
```

Comment:
```suggestion
static s16 randomShieldDiscount[] = { 5, 10, 15, 20, 25, 30, 35, 40 };
```
Can probably go with something like this, and obviously replacing the other uses of this.